compiler/rustc_span/src/source_map.rs

```diff
@@ -115,6 +115,10 @@ impl FileLoader for RealFileLoader {
     }
 
     fn read_file(&self, path: &Path) -> io::Result<String> {
+        if path.metadata().is_ok_and(|metadata| metadata.len() > u32::MAX.into()) {
```

Comment:
Could we report this as an `io::Error::other` rather than fataling here?

Reply:
Yes, this works and produces:
```
error: couldn't read /home/tyilo/tmp/big_source.rs: rustc does not support files larger than 4 GiB

error: aborting due to 1 previous error
```